internal/cortex/chunk/encoding/bigchunk.go

```diff
@@ -278,7 +278,7 @@ func (it *bigchunkIterator) FindAtOrAfter(target model.Time) bool {
 		it.curr = it.chunks[it.i].Iterator(it.curr)
 	}
 
-	for it.curr.Next() {
+	for it.curr.Next() != chunkenc.ValNone {
```

Comment:
Do we also need to skip over histograms (`chunkenc.ValHistogram`, `chunkenc.ValFloatHistogram`) here? :thinking: same in other places.

Reply:
Currently we don't support ingesting native histograms (they are ignored see types) and we can ignore them from what I understand. The plan is to add this in a follow up PR. Idea of this PR is to prepare for that by updating to Prometheus v0.40.1.

Reply:
Actually those part of code was removed on Cortex already since Cortex dropped support for chunk store.
I am wondering how should we deal with the vendored Cortex code... 
It is bad to maintain the deprecated code and it is a pain when updating Prometheus

Reply:
I think we should remove unused code as soon as possible. Not sure if this PR is the best time to do it, but maybe we can have a follow up before adding support for native histograms.

Reply:
I created an issue for that: https://github.com/thanos-io/thanos/issues/5912

Maybe we can have the discussion there and I agree with you and think it might be a good idea to do that before starting work on native histograms.

Reply:
I'd agree with a followup before implementing native histograms. For now having as it this in this PR is fine for me.